fe-library-app/src/components/Search/Search.tsx

```diff
@@ -1,9 +1,26 @@
+import { useEffect, useState } from 'react'
+
 import styles from './Search.module.css'
 
-const Search = () => {
+interface SearchProps{
+  searchChange: (newInput: string) => void
+}
+
+const Search = (props: SearchProps) => {
+  const [ enteredSearch, setEnteredSearch ] = useState('')
+
+  useEffect(() => {
+    const timeOut = setTimeout(() => props.searchChange(enteredSearch), 500)
+    return () => clearTimeout(timeOut)
```

Comment:
If I'm not mistaken, this will still execute search on every keystroke, just it will be delayed. Use debounce instead.

Reply:
Test this statement firstly

Reply:
https://stackoverflow.com/questions/66394801/execute-api-request-when-user-stops-typing-search-box

Reply:
It's tested it works with timeout